File: src/main/java/org/sonar/plugins/xquery/checks/AbstractProhibitStringValueCheck.java
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.checks;

import org.sonar.plugins.xquery.parser.XQueryParser;
import org.sonar.plugins.xquery.parser.XQueryTree;

import java.util.regex.Pattern;

public abstract class AbstractProhibitStringValueCheck extends AbstractCheck {
      
    @Override
    public void enterExpression(XQueryTree node) {
        super.enterExpression(node);
        
        if (XQueryParser.StringLiteral == node.getType())
        {
    	   String value = node.getValue();
    	   if (Pattern.matches(getPattern(), value)) {
    		   createIssue(node.getLine());
    	   }
        }
    }

    abstract protected void createIssue(int lineNumber);
    abstract protected String getPattern();
}

File: src/main/java/org/sonar/plugins/xquery/checks/AbstractProhibitStringValueCheck.kt
/*
 * © 2014 by Intellectual Reserve, Inc. All rights reserved.
 */

package org.sonar.plugins.xquery.checks

import org.antlr.v4.runtime.ParserRuleContext
import org.sonar.plugins.xquery.parser.XQueryParser
import org.sonar.plugins.xquery.parser.getLine
import java.util.regex.Pattern

abstract class AbstractProhibitStringValueCheck : AbstractCheck() {
    protected abstract val pattern: String

    override fun enterExpression(node: ParserRuleContext) {
        super.enterExpression(node)

        if (node is XQueryParser.StringLiteralContext) {
            val value = node.text
            if (Pattern.matches(pattern, value)) {
                createIssue(node.getLine())
            }
        }
    }

    protected abstract fun createIssue(lineNumber: Int)
}
